### PR TITLE
Enable Windows build

### PR DESCRIPTION
**Pull request by seanyen**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

* Added `seanyen` as a co-maintainer. Hope that's okay. :)
* Patched `pytohn/CMakeLists.txt` to define an alternative option `EIGENPY_STIELIB_ROOT` to install the `Python` modules to a different location. It is to accommodate `Conda` on `Windows` convention where all the native binaries go to `%PREFIX%\Library` but all `Python` things go to `%PREFIX\lib`.
* Patched the `Eigen3` package discovery because on Windows, it doesn't export `PkgConfig` file yet.

### CODE REVIEW

**Comment by seanyen** [comment]
Link a related pull request: https://github.com/conda-forge/eigen-feedstock/pull/24

**Comment by seanyen** [comment]
cc @conda-forge/eigenpy this is ready for review and merge. thanks!

**Comment by lesteve** [comment]
I have to admit I don't know how to review this, since I am not a conda-forge expert and know very little about Windows.

@seanyen what do you recommend, merging and fixing problems later if they are reported?

cc @wolfv if he has a more informed opinion.

File: recipe/windows.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@@ -26,13 +26,17 @@
+ ADD_LIBRARY(${PYWRAP} SHARED main.cpp)
+ TARGET_LINK_LIBRARIES(${PYWRAP} ${PROJECT_NAME}) 
+ TARGET_LINK_BOOST_PYTHON(${PYWRAP})
+ #IF(BUILD_WITH_COMMIT_VERSION)
+ #  TAG_LIBRARY_VERSION(${PYWRAP})
+ #ENDIF(BUILD_WITH_COMMIT_VERSION)
+-SET(${PYWRAP}_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/${PYTHON_SITELIB})
++IF(DEFINED EIGENPY_STIELIB_ROOT AND EIGENPY_STIELIB_ROOT)
++  SET(${PYWRAP}_INSTALL_DIR ${EIGENPY_STIELIB_ROOT}/${PYTHON_SITELIB})
```
Comment (wolfv) [inline recipe/windows.patch]: Is there a typo in eigenpy_sitelib?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,16 +8,20 @@ package:
...
 build:
   number: 0
-  skip: true  # [win]
+  skip: true  # [win and vc<14]
 
 requirements:
   build:
-    - {{ compiler('cxx') }}
+    - {{ compiler('cxx') }}  # [not win]
+    - vs2017_win-64  # [win64]
```
Comment (wolfv) [inline recipe/meta.yaml]: Shouldn’t the compiler('cxx') stuff also work for win?

Comment (wolfv) [reply recipe/meta.yaml]: I asked on the gitter channel and apparently it doesn't work if you want vs2017 (as opposed to 2014) which I guess is intended here :)

**Comment by wolfv** [comment]
Everything looks good to me except for the (typo?) in `EIGENPY_STIELIB_ROOT`

I quickly searched the eigenpy repo and didn't find this variable, spelled either correclty or wrong. Not sure from where it comes :) if you tell me it's correct like this i am happy to merge this recipe!

**Comment by seanyen** [comment]
> Everything looks good to me except for the (typo?) in `EIGENPY_STIELIB_ROOT`

Thanks for the fix. That's indeed a typo! And the variable is something made-up by me since the original build recipe assumes all the binaries should go under the same prefix, but it is not the case for `conda` on Windows.

File: recipe/windows.patch (hunk 1 above)
Comment (seanyen) [reply recipe/windows.patch]: Thanks for the typo fix!

File: recipe/meta.yaml (hunk 2 above)
Comment (seanyen) [reply recipe/meta.yaml]: Thanks for the confirmation. I will keep it as is. Btw, I learned this trick from `qt-feedstock`. :)

File: recipe/bld.bat [hunk 3]
```diff
@@ -0,0 +1,20 @@
+mkdir build
+cd build
+
+cmake ^
+    -G "NMake Makefiles" ^
+    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
```
Comment (jcarpent) [inline recipe/bld.bat]: It seems that this variable is only defined for Win32 systems (according to https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html)

**Pull request merged**